### PR TITLE
Fix fullscreen resolution

### LINKED ISSUE #1294: [core] ToggleFullscreen() could change screen width/height
When calling `ToggleFullscreen()`, it could (wrongly) change requested `screenWidth` or `screenHeight`.

Internal call sequence go as follow: 
```c
ToggleFullScreen() 
    glfwSetWindowMonitor()      // This function calculate the closest monitor-supported width x height
        WindowSizeCallback()    // This function is called with a wrong width or height
            SetupViewport()     // Setup viewport and projection matrix 
```

Maybe issue is GLFW side and `glfwSetWindowMonitor()` shouldn't call `WindowSizeCallback()`?

EDIT 1: One possible solution is just commenting `WindowSizeCallback()` internals.

EDIT 2: `InitGraphicsDevice()` also looks for closest monitor supported `width x height` and modifies it.

### PR DESCRIPTION
**Pull request by object71**
Fixes #1294
Uses not yet submitted pull request of @JeffM2501 - #1636 

Modified how fullscreen gets toggled.

- Removed the unsetting and setting of the resize callback function. Instead of that I moved the fullscreen flag setting into a more correct place before setting the fullscreen so that this flag can be used in the callback.
- In the resize callback now window size is only set when it is not fullscreen resulting in preserving the window size.
- When going fullscreen the larges resolution is used so that there are no problems of the type when you minimize the window you cannot use anything else in your desktop because the resolution might be too low. If a low res effect is desired one should use render texture (this is the approach all game engines use).

@raysan5 This should yield the required results.

### CODE REVIEW

**Comment by JeffM2501** [comment]
The main reason I did not do this in my PR was that I was not 100% sure if the intent of the function was to change resolutions or not.

I feel that it is silly to change resolutions on modern displays, so I completely understand the desire for this change.
My issue is that this change removes the ability to change the display resolution. I would add a SetMonitorResolution function to this PR so that the functionality is not lost.

I agree that ToggleFullscreen should work this way, but I would hate for raylib to loose functionality for someone who may have a true need to change monitor resolutions.

**Comment by object71** [comment]
@JeffM2501 Looking at documentation people can just use SetWindowSize after fullscreen is enabled to set the resolution that they want.

**Comment by object71** [comment]
Also if people want to have a low-res effect they can go on for full resolution and just have a texture render target as shown in the example core_window_letterbox which will be rescaled to the new fullscreen size

File: src/core.c [hunk 1]
```diff
@@ -1034,34 +1034,41 @@ void ToggleFullscreen(void)
...
+        
         const GLFWvidmode *mode = glfwGetVideoMode(monitor);
-        glfwSetWindowSizeCallback(CORE.Window.handle, NULL);
-        glfwSetWindowMonitor(CORE.Window.handle, monitor, 0, 0, CORE.Window.screen.width, CORE.Window.screen.height, GLFW_DONT_CARE);
-        glfwSetWindowSizeCallback(CORE.Window.handle, WindowSizeCallback);
-
-        // Try to enable GPU V-Sync, so frames are limited to screen refresh rate (60Hz -> 60 FPS)
-        // NOTE: V-Sync can be enabled by graphic driver configuration
-        if (CORE.Window.flags & FLAG_VSYNC_HINT) glfwSwapInterval(1);
+        glfwSetWindowMonitor(CORE.Window.handle, monitor, 0, 0, mode->width, mode->height, mode->refreshRate);
```
Comment (raysan5) [inline src/core.c:1059]: Actually, fullscreen mode shouldn't change monitor video mode resolution, it should try to find the closest resolution supported by the monitor to the user requested one.

**Comment by raysan5** [comment]
@object71 Added one comment. Fullscreen switch shouldn't change monitor resolution but try to find the closest monitor resolution available for the user desired one. Actually, that's the problematic part, for example:

```
User game resolution:  800x450 (16:9)
Fullscreen toggle resolution selected:  800x600 (4:3) - the closest one to a supported monitor resolution.
```

What should happen in that case is that we keep using a 800x450 "game" resolution but add an offsetY with black bars to the rendering... `SetupFramebuffer()` tries to do that...

For user selected "standard" resolutions (1280x720, 1920x1080...), fullscreen works as expected

Probably the best option at this point is just moving that part of the work to user side and let the user create and manage a custom `RenderTexture` like in `core_window_letterbox` example.

Another issue also related to fullscreen is https://github.com/raysan5/raylib/issues/1241 but that's related to browser behaviour and security concerns while switching fullscreen from application side...

**Comment by object71** [comment]
@raysan5 So do I do something in this PR? Remove the setting fullscreen with the max video mode and use the closest to the initial window size?

**Comment by raysan5** [comment]
@object71 Yes, we can just keep the desired user resolution and let the monitor respond to it.

**Comment by object71** [comment]
Okay, resolving that in a moment

**Comment by raysan5** [comment]
@object71 Nice, keeping an eye! I'll merge when ready! :)

**Comment by object71** [comment]
@object71 Done

**Comment by raysan5** [comment]
@object71 Merged!

**Pull request merged**